fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -154,42 +141,91 @@ private void parseFieldNames(KeyExpression expression, @Nonnull String parentFie
                                                                           final boolean remove,
                                                                           @Nonnull final List<IndexEntry> indexEntries) {
         LOG.trace("updateIndexKeys savedRecord={}, remove=={}, indexEntries={}", savedRecord, remove, indexEntries);
+
         if (indexEntries.isEmpty()) {
             return AsyncUtil.DONE;
         }
         try {
             final IndexWriter writer = getOrCreateIndexWriter(state, analyzer, executor);
+            Document document = new Document();
+            byte[] primaryKey = savedRecord.getPrimaryKey().pack();
+            BytesRef ref = new BytesRef(primaryKey);
+            document.add(new StoredField(PRIMARY_KEY_FIELD_NAME, ref));
+            document.add(new SortedDocValuesField(PRIMARY_KEY_SEARCH_NAME, ref));
```

Comment:
I could imagine wanting to store a multi-field primary-key as _separate_ document fields, so as to be able to query on just some of the components.

Reply:
I guess thats a question of design too, should you have to specify the fields in the index definition that you want to query on or should we assume primary field keys are included.

Reply:
I am guessing that we will need some level of control over how you intend the index to be used, though with sensible defaults.